lib/core/mixins/login_screen_mixin.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:asset_tracker/core/constants/strings/locale/tr_strings.dart';
+import 'package:asset_tracker/core/exceptions/firebase_auth_exceptions/firebase_error_type.dart';
+import 'package:asset_tracker/core/exceptions/firebase_auth_exceptions/firebase_exception.dart';
+import 'package:asset_tracker/core/extensions/build_context_extension.dart';
+import 'package:asset_tracker/core/extensions/firebase_error_extension.dart';
+import 'package:asset_tracker/core/extensions/snack_bar_extension.dart';
+import 'package:asset_tracker/core/routing/route_names.dart';
+import 'package:asset_tracker/features/auth/presentation/state_management/auth_state_manager.dart';
+import 'package:asset_tracker/features/auth/presentation/state_management/auth_state_provider.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+mixin LoginScreenMixin {
+  final formKey = GlobalKey<FormState>();
+  final emailController = TextEditingController();
+  final passwordController = TextEditingController();
+
+  Text signInTextTitle(BuildContext context) {
+    return Text(
+      TrStrings.signIn,
+      style: context.textTheme.headlineLarge
+          ?.copyWith(color: context.colorScheme.onSurface),
+    );
+  }
+
+  ElevatedButton customElevatedButton(BuildContext context, WidgetRef ref) {
+    return ElevatedButton(
```

Comment:
Bu buton login için özelleştirilebilir. LoginButton şeklinde, dışarıdan future fonksiyon alır. vs vs